pkg/destroy/gcp/compute.go

```diff
@@ -0,0 +1,225 @@
+	result := []nameAndZone{}
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+	req := o.computeSvc.InstanceGroups.AggregatedList(o.ProjectID).Fields("items/*/instanceGroups(name,zone)").Filter(filter)
+	err := req.Pages(ctx, func(list *compute.InstanceGroupAggregatedList) error {
+		for _, scopedList := range list.Items {
+			for _, ig := range scopedList.InstanceGroups {
+				zoneName := o.getZoneName(ig.Zone)
+				result = append(result, nameAndZone{
+					name: ig.Name,
+					zone: zoneName,
+				})
+				o.Logger.Debugf("Found instance group %s in zone %s\n", ig.Name, zoneName)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to fetch compute instance groups")
+	}
+	return result, nil
+}
+
+func (o *ClusterUninstaller) listInstanceGroupInstances(ig nameAndZone) ([]nameAndZone, error) {
+	o.Logger.Debugf("Listing instance group instances for %v", ig)
+	result := []nameAndZone{}
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+	req := o.computeSvc.InstanceGroups.ListInstances(o.ProjectID, ig.zone, ig.name, &compute.InstanceGroupsListInstancesRequest{}).Fields("items(instance)")
+	err := req.Pages(ctx, func(list *compute.InstanceGroupsListInstances) error {
+		for _, item := range list.Items {
+			name, zone := o.getInstanceNameAndZone(item.Instance)
+			if len(name) == 0 {
+				continue
+			}
+			result = append(result, nameAndZone{
+				name: name,
+				zone: zone,
+			})
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to fetch instance group instances")
+	}
+	return result, nil
+}
+
+func (o *ClusterUninstaller) deleteInstanceGroup(ig nameAndZone) error {
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+	o.Logger.Debugf("Deleting instance group %s in zone %s", ig.name, ig.zone)
+	_, err := o.computeSvc.InstanceGroups.Delete(o.ProjectID, ig.zone, ig.name).Context(ctx).Do()
+	if err != nil && !isNoOp(err) {
+		return errors.Wrapf(err, "failed to delete instance group %s in zone %s", ig.name, ig.zone)
+	}
+	return nil
+}
+
+func (o *ClusterUninstaller) destroyInstanceGroups() error {
+	instanceGroups, err := o.listClusterInstanceGroups()
+	if err != nil {
+		return err
+	}
+	for _, ig := range instanceGroups {
+		err := o.deleteInstanceGroup(ig)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *ClusterUninstaller) listComputeInstances() ([]nameAndZone, error) {
+	o.Logger.Debugf("Listing compute instances")
+	result := []nameAndZone{}
+	filter := fmt.Sprintf("(status ne TERMINATED)(%s)", o.clusterIDFilter())
+	req := o.computeSvc.Instances.AggregatedList(o.ProjectID).Filter(filter).Fields("items/*/instances(name,zone,status)")
+	ctx, cancel := o.contextWithTimeout()
+	defer cancel()
+	err := req.Pages(ctx, func(list *compute.InstanceAggregatedList) error {
+		for _, scopedList := range list.Items {
+			for _, instance := range scopedList.Instances {
+				if !o.isClusterResource(instance.Name) {
```

Comment:
This shouldn't be required with change `filter := fmt.Sprintf("(status ne TERMINATED)(%s)", o.clusterIDFilter())` https://github.com/openshift/installer/pull/1950/commits/3f79cb68375cf9be05d7aba8403d7771917ea65d#diff-47aedd2bb0ff065c4f42429f4963e077R130 right?